packages/lesswrong/lib/collections/posts/helpers.ts

```diff
@@ -155,15 +156,34 @@ export const postGetLastCommentPromotedAt = (post: PostsBase|DbPost):Date|null =
   return post.lastCommentPromotedAt;
 }
 
-export const postCanEdit = (currentUser: UsersCurrent|DbUser|null, post: PostsBase|DbPost): boolean => {
-  return userOwns(currentUser, post) || userCanDo(currentUser, 'posts.edit.all')
+/**
+ * Whether or not the given user is an organizer for the post's group
+ * @param user
+ * @param post
+ * @returns {Promise} Promise object resolves to true if the post has a group and the user is an organizer for that group
+ */
+export const userIsPostGroupOrganizer = async (user: UsersMinimumInfo|DbUser|null, post: PostsBase|DbPost): Promise<boolean> => {
+  const groupId = ('group' in post) ? post.group?._id : post.groupId;
+  if (!user || !groupId)
+    return false
+    
+  const group = await Localgroups.findOne({_id: groupId});
+  return !!group && group.organizerIds.some(id => id === user._id);
 }
 
-export const postCanDelete = (currentUser: UsersCurrent|DbUser|null, post: PostsBase|DbPost): boolean => {
+export const postCanEdit = (currentUser: UsersCurrent|null, post: PostsBase): boolean => {
+  const organizerIds = post.group?.organizerIds;
+  const isPostGroupOrganizer = organizerIds ? organizerIds.some(id => id === currentUser?._id) : false;
+  return userOwns(currentUser, post) || userCanDo(currentUser, 'posts.edit.all') || isPostGroupOrganizer;
+}
+
+export const postCanDelete = (currentUser: UsersCurrent|null, post: PostsBase): boolean => {
```

Comment:
Note that `postCanDelete` is effectively dead code; it might be reachable via the GraphQL API, but the site UI never deletes posts (or deletes anything, really), it just soft-deletes stuff by setting `deleted:true` (most things) or `draft:true` and `deletedDraft:true` (posts in particular).